components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/SettingsMappingUtil.java

```diff
@@ -49,8 +48,16 @@ public SettingsDTO fromSettingstoDTO(Boolean isUserAvailable, Boolean moneatizat
             settingsDTO.setApplicationSharingEnabled(APIUtil.isMultiGroupAppSharingEnabled());
             settingsDTO.setMapExistingAuthApps(APIUtil.isMapExistingAuthAppsEnabled());
             Map<String, Environment> environments = APIUtil.getEnvironments();
-            Environment environment = environments.get(RestApiConstants.DEFAULT_ENVIRONMENT);
-            settingsDTO.apiGatewayEndpoint(environment.getApiGatewayEndpoint());
+            for (Map.Entry<String, Environment> entry : environments.entrySet()) {
+                Environment multipleEnvironment = environments.get(entry.getKey());
+                if (multipleEnvironment.isDefault()) {
+                    settingsDTO.apiGatewayEndpoint(multipleEnvironment.getApiGatewayEndpoint());
+                    break;
+                }
+            }
+            Map.Entry<String, Environment> entry = environments.entrySet().iterator().next();
+            Environment singleEnvironment = environments.get(entry.getKey());
+            settingsDTO.apiGatewayEndpoint(singleEnvironment.getApiGatewayEndpoint());
```

Comment:
aren't we replacing the value added at line 54 by this one here?

Reply:
Fixed with 1bd989f